File: LeetCode/3회차/B04_Most Common Word.py
from typing import List
from collections import Counter


class Solution:
    def mostCommonWord(self, paragraph: str, banned: List[str]) -> str:
        paragraph = paragraph.lower()
        for char in paragraph:
            if ord(char) < 97 or ord(char) > 122:
                paragraph = paragraph.replace(char, ' ')

        counts = Counter(paragraph.split())
        for word, count in counts.most_common():
            if word not in banned:
                return word


print(Solution().mostCommonWord("Bob hit a ball, the hit BALL flew far after it was hit.",
["hit"]))

File: LeetCode/JavaScript/B04-Most Common Word.js
/**
 * @param {string} paragraph
 * @param {string[]} banned
 * @return {string}
 */
var mostCommonWord = function (paragraph, banned) {
  const words = paragraph
    .toLowerCase()
    .replace(/[^a-z]/g, " ")
    .split(" ");
  const counts = new Map();

  for (word of words) {
    if (word && !banned.includes(word)) {
      const count = counts.get(word);
      if (count) {
        counts.set(word, count + 1);
      } else {
        counts.set(word, 1);
      }
    }
  }
  return Array.from(counts.entries()).sort((a, b) => (a[1] < b[1] ? 1 : -1))[0][0];
};

console.log(mostCommonWord("Bob hit a ball, the hit BALL flew far after it was hit.", ["hit"]));
